ambry-tools/src/main/java/com.github.ambry/tools/admin/ConcurrencyTestTool.java

```diff
@@ -0,0 +1,975 @@
+ * --burstThreadCountForPut int         : No of concurrent PUTs by each producer thread
+ */
+public class ConcurrencyTestTool {
+  private static final Logger logger = LoggerFactory.getLogger(ConcurrencyTestTool.class);
+  private static final String ROUTER_PUT_GET_HELPER = "com.github.ambry.tools.admin.RouterPutGetHelperFactory";
+  private static final String SERVER_PUT_GET_HELPER = "com.github.ambry.tools.admin.ServerPutGetHelperFactory";
+
+  public static void main(String args[])
+      throws Exception {
+    InvocationOptions options = new InvocationOptions(args);
+    Properties properties = Utils.loadProps(options.routerPropsFilePath);
+    ClusterMap clusterMap = new ClusterMapManager(options.hardwareLayoutFilePath, options.partitionLayoutFilePath,
+        new ClusterMapConfig(new VerifiableProperties(properties)));
+
+    PutGetHelperFactory putGetHelperFactory = null;
+    if (options.putGetHelperFactoryStr.equals(ROUTER_PUT_GET_HELPER)) {
+      putGetHelperFactory =
+          Utils.getObj(options.putGetHelperFactoryStr, properties, clusterMap, options.maxBlobSizeInBytes,
+              options.minBlobSizeInBytes);
+    } else if (options.putGetHelperFactoryStr.equals(SERVER_PUT_GET_HELPER)) {
+      putGetHelperFactory =
+          Utils.getObj(options.putGetHelperFactoryStr, properties, options.hostName, options.port, clusterMap,
+              options.maxBlobSizeInBytes, options.minBlobSizeInBytes);
+    }
+    PutGetHelper putGetHelper = putGetHelperFactory.getPutGetHelper();
+    ConcurrencyTestTool concurrencyTestTool = new ConcurrencyTestTool();
+    concurrencyTestTool.startTest(putGetHelper, options.maxParallelPutCount, options.parallelGetCount,
+        options.totalPutBlobCount, options.maxGetCountPerBlob, options.burstCountForGet,
+        options.maxFailuresPerPutBatchToStopPuts, options.maxFailuresPerGetBatchToStopGets, options.deleteAndValidate,
+        options.sleepTimeBetweenBatchPutsInMs, options.sleepTimeBetweenBatchGetsInMs);
+  }
+
+  /**
+   * Initializes and starts the concurrent test tool.
+   * @param putGetHelper the {@link PutGetHelper} to be used for PUTs and GETs
+   * @param maxParallelPutCount maximum number of parallel puts per phase
+   * @param parallelGetCount the number of parallel Gets per phase
+   * @param totalPutCount total number of blobs to be uploaded
+   * @param maxGetCountPerBlob total number of times a particular blob can be fetched before its deleted
+   * @param burstCountGet Burst count for GETs. Each GET consumer thread will fetch a blob these many no of
+   *                         times simultaneously
+   * @param maxFailuresPerPutBatchToStopPuts Maximum allowable failures per Put batch in order to continue with PUTs
+   * @param maxFailuresPerGetBatchToStopGets Maximum allowable failures per Get batch in order to continue with GETs
+   * @param deleteAndValidate Deletes the blobs once GETs have reached a threshold on every blob
```

Comment:
Yes, but lets say there is investigation going on and so we may need the blobs to be available and not deleted during such cases.